tardis/montecarlo/src/integrator.c

```diff
@@ -336,8 +337,8 @@ _formal_integral(
           }
         }
       // Free everything allocated on heap
-      free(exp_tau);
       printf("\n");
     }
+      free(exp_tau);
```

Comment:
I think this is an unintended consequence - did you use the newest upstream/master?